src/interface.ts

```diff
@@ -80,9 +89,9 @@ interface DesktopAgent {
   broadcast(context: Context): void;
 
   /**
-   * Constructs a new intent
+   * Raises an intent to the desktop agent to resolve.
    */
-  intent(intent: IntentName, context: Context, target: String): Intent;
+  raiseIntent(intent: IntentName, context: Context, target: String): Promise<IntentResolution>;
```

Comment:
If we are switching to `raiseIntent` as a function call (which I am definitely in favour of, see #12), we should remove the Intent object definition above, as that also has a `send` method, and a context, which conflicts with this/introduces ambigiuty.

Please see https://github.com/rikoe/API/pull/1 for my version of the same change, which is very similar (I am not able to raise a PR to upstream as I am not part of a FINOS CLA).